docs/migration_guide.md

```diff
@@ -0,0 +1,127 @@
+# Migration guide from YAML payloads to JSON (bfb) protocol
+
+## Background
+Communication protocol between SUA and backend was extended to support vehicle orchestration for large quantities of devices. Key differences are:
+* Transition from YAML to JSON
+* Fine-grained API (command-based approach instead of single-shot self-update)
+* Extension of statuses (for detailed reflection of success, progress or failure)
+Some fiels are not relevant anymore meanwhile new were introduced. Below is a brief comparison between YAML and JSON messages.
+
+### Transition guide for current state message
+From the current state message in YAML payload there is only 1 field 'bundleVersion' which is relevant and corresponding field in JSON is 'version' in 'softwareNodes' for the device image (actual name could differ depending on distro configuration):
+```
+apiVersion: sdv.eclipse.org/v1
+kind: SelfUpdateBundle
+metadata: 
+    name: "self-update-bundle-example"
+spec: 
+    bundleVersion: 1.0
+```
+
+```
+{
+    "timestamp": 42,
+    "payload": {
+        "softwareNodes": [
+            {
+                "id": "self-update-agent",
+                "version": "build-42",
+                "name": "OTA NG Self Update Agent",
+                "type": "APPLICATION"
+            },
+            {
+                "id": "self-update:leda-deviceimage",
+                "version": "1.0",
+                "name": "Official Leda device image",
+                "type": "IMAGE"
+            }
+        ],
+        "hardwareNodes": [],
+        "associations": [
+            {
+                "sourceId": "self-update-agent",
+                "targetId": "self-update:leda-deviceimage"
+            }
+        ]
+    }
+}
+```
+
+### Transition guide for desired state message
+For the desired state message there are 2 relevant fields 'bundleDownloadUrl' and 'bundleVersion' for take-over. The corresponding values are 'version' under the 'components' and 'value' inside the object with 'key'='image'. *NOTE*: There could be multiple entries inside 'components' array and 'config' section. Your implementation has to search for 'os-image' and 'image' values respectively.
+```
+apiVersion: "sdv.eclipse.org/v1"
+kind: SelfUpdateBundle
+metadata:
+    name: self-update-bundle-example
+spec:
+    bundleDownloadUrl: http://url
+    bundleName: arm64-bundle
+    bundleTarget: base
+    bundleVersion: 1.0
```

Comment:
Is it possible to highlight the relevant fields?

Reply:
No. Markdown for preformatted text will supersede all HTML markup inside the block.